Samples/StyleTransfer/VideoEffect/StyleTransferEffectCpp/StyleTransferEffect.cpp

```diff
@@ -0,0 +1,94 @@
+﻿#include "pch.h"
+#include "StyleTransferEffect.h"
+#include "StyleTransferEffect.g.cpp"
+using namespace std;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+
+namespace winrt::StyleTransferEffectCpp::implementation
+{
+	StyleTransferEffect::StyleTransferEffect() : outputTransformed(VideoFrame(Windows::Graphics::Imaging::BitmapPixelFormat::Bgra8, 720, 720)),
+		Session(nullptr),
+		Binding(nullptr)
+	{
+	}
+
+	IVectorView<VideoEncodingProperties> StyleTransferEffect::SupportedEncodingProperties() {
+		VideoEncodingProperties encodingProperties = VideoEncodingProperties();
+		encodingProperties.Subtype(L"ARGB32");
+		return single_threaded_vector(std::move(std::vector<VideoEncodingProperties>{encodingProperties})).GetView();
+	}
+
+	bool StyleTransferEffect::TimeIndependent() { return true; }
+	MediaMemoryTypes StyleTransferEffect::SupportedMemoryTypes() { return MediaMemoryTypes::GpuAndCpu; }
+	bool StyleTransferEffect::IsReadOnly() { return false; }
+	void StyleTransferEffect::DiscardQueuedFrames() {}
+
+	void StyleTransferEffect::Close(MediaEffectClosedReason m) {
+		OutputDebugString(L"Close Begin | ");
+		std::lock_guard<mutex> guard{ Processing };
+		OutputDebugString(L"Close\n");
+		if (Binding != nullptr) Binding.Clear();
+		if (Session != nullptr) Session.Close();
+		outputTransformed.Close();
+	}
+
+
+	void StyleTransferEffect::ProcessFrame(ProcessVideoFrameContext context) {
+		auto now = std::chrono::high_resolution_clock::now();
+		std::chrono::milliseconds timePassed;
+		// If the first time calling ProcessFrame, just start the timer 
+		if (firstProcessFrameCall) {
+			m_StartTime = now;
+			firstProcessFrameCall = false;
+		}
+		// On the second and any proceding process, 
+		else {
+			timePassed = std::chrono::duration_cast<std::chrono::milliseconds>(now - m_StartTime);
+			m_StartTime = now;
+			Notifier.SetFrameRate(1000.f / timePassed.count()); // Convert to FPS: milli to seconds, invert 
+		}
+
+		OutputDebugString(L"PF Start | ");
+
+		VideoFrame inputFrame = context.InputFrame();
+		VideoFrame outputFrame = context.OutputFrame();
+
+		std::lock_guard<mutex> guard{ Processing };
+		OutputDebugString(L"PF Locked | ");
+		Binding.Bind(InputImageDescription, inputFrame);
+		Binding.Bind(OutputImageDescription, outputTransformed);
+
+		OutputDebugString(L"PF Eval | ");
+		Session.Evaluate(Binding, L"test");
+		outputTransformed.CopyToAsync(outputFrame).get();
```

Comment:
Is `CopyToAsync` doing any transformations? If not you could put it directly into context.OutputFrame()

Reply:
I use CopyToAsync for auto-cropping, since the output from Session.Evaluate is a VideoFrame of size 720x720, and context.OutputFrame() is a VideoFrame of size 640x360. I'm taking a second stab at whether I can resize the mediacapture itself to fit the dimensions of the model.

Reply:
Is this the behavior you see? VideoFrame.CopyToAsync doesn't seem to resize:

> The target frame must have the same pixel format, width, height, and alpha mode as the source frame.

(from https://docs.microsoft.com/en-us/uwp/api/windows.media.videoframe.copytoasync?view=winrt-19041 . Could be that the docs are outdated though)